src/main/java/org/sunbird/assessment/service/AssessmentServiceV2Impl.java

```diff
@@ -311,89 +489,58 @@ private Map<String, Object> filterQuestionMapDetail(Map<String, Object> question
-        sectionLevelResult.put(Constants.INCORRECT, resultMap.get(Constants.INCORRECT));
-        sectionLevelResult.put(Constants.PASS, result >= passPercentage);
-
-        List<Map<String, Object>> sectionChildren = new ArrayList<Map<String, Object>>();
-        sectionChildren.add(sectionLevelResult);
-        outgoingResponse.getResult().put(Constants.CHILDREN, sectionChildren);
+        Double result;
+        if (resultMap != null) {
+            result = (Double) resultMap.get(Constants.RESULT);
+            sectionLevelResult.put(Constants.RESULT, result);
+            sectionLevelResult.put(Constants.TOTAL, resultMap.get(Constants.TOTAL));
+            sectionLevelResult.put(Constants.BLANK, resultMap.get(Constants.BLANK));
+            sectionLevelResult.put(Constants.CORRECT, resultMap.get(Constants.CORRECT));
+            sectionLevelResult.put(Constants.INCORRECT, resultMap.get(Constants.INCORRECT));
+        } else {
+            result = 0.0;
+            sectionLevelResult.put(Constants.RESULT, result);
+            List<String> childNodes = (List<String>) hierarchySection.get(Constants.CHILDREN);
+            sectionLevelResult.put(Constants.TOTAL, childNodes.size());
+            sectionLevelResult.put(Constants.BLANK, childNodes.size());
+            sectionLevelResult.put(Constants.CORRECT, 0);
+            sectionLevelResult.put(Constants.INCORRECT, 0);
+        }
+        sectionLevelResult.put(Constants.PASS, result >= ((Integer) hierarchySection.get(Constants.MINIMUM_PASS_PERCENTAGE)));
+        sectionLevelResult.put(Constants.OVERALL_RESULT, result);
+        return sectionLevelResult;
+    }
 
-        outgoingResponse.getResult().put(Constants.OVERALL_RESULT, result);
-        outgoingResponse.getResult().put(Constants.TOTAL, resultMap.get(Constants.TOTAL));
-        outgoingResponse.getResult().put(Constants.BLANK, resultMap.get(Constants.BLANK));
-        outgoingResponse.getResult().put(Constants.CORRECT, resultMap.get(Constants.CORRECT));
-        outgoingResponse.getResult().put(Constants.PASS_PERCENTAGE, hierarchySection.get(Constants.MINIMUM_PASS_PERCENTAGE));
-        outgoingResponse.getResult().put(Constants.INCORRECT, resultMap.get(Constants.INCORRECT));
-        outgoingResponse.getResult().put(Constants.PASS, result >= passPercentage);
+    private SBApiResponse createDefaultResponse(String api) {
+        SBApiResponse response = new SBApiResponse();
+        response.setId(api);
+        response.setVer(Constants.VER);
+        response.getParams().setResmsgid(UUID.randomUUID().toString());
+        response.getParams().setStatus(Constants.SUCCESS);
+        response.setResponseCode(HttpStatus.OK);
+        response.setTs(DateTime.now().toString());
+        return response;
     }
 
+    private Boolean validateQuestionListRequest(List<String> identifierList, List<String> questionsFromAssessment) {
```

Comment:
This method should return String (i.e. errMsg) -- in readQuestionList just check errMsg is not empty -- if so, return error response...